### PR TITLE
Add font Zed Mono

### PR DESCRIPTION
**Pull request by cabrinha**
#### Description

Added a new font: "Zed Mono"

#### Requirements / Checklist

- [x] Read the [Contributing Guidelines](https://github.com/ryanoasis/nerd-fonts/blob/-/contributing.md)
- [x] Verified the license of any newly added font, glyph, or glyph set

#### What does this Pull Request (PR) do?

Adds a new font

#### Any background context you can provide?

https://zed.dev/

https://github.com/zed-industries/zed-fonts/issues/10

#### Screenshots (if appropriate or helpful)

Screenshot 2024-02-01 at 6 38 35 PM

### CODE REVIEW

**Comment by Blankeos** [comment]
@cabrinha Thanks for working on this! Still waiting for this to be released haha. Zed Mono has become my favorite font. I'm planning to just use this PR and install the fonts from there. Do I need to run anything (run a build or something) or will the patched `.ttf` work out-of-the-box?

**Comment by arlyon** [comment]
I downloaded this and don't see any difference between iosevka.

Screenshot 2024-03-15 at 16 24 28

Screenshot 2024-03-15 at 16 24 34

Screenshot 2024-03-15 at 16 24 42

**Comment by Blankeos** [comment]
@arlyon So it works. Even works for the native MacOS Terminal (It has no fallback fonts unlike VSCode). Also got it to work on VSCode.

By the way, `"Zed Mono"` by itself does look 'weird' in my opinion because it looks too thin. I prefer the "Extended" version because it makes it have almost the character spacing as JetBrains.

image

### Steps I did to get this:

1. I clicked on @cabrinha's remote forked repo.
2. I went to [nerd-fonts](https://github.com/cabrinha/nerd-fonts/tree/zed-fonts)/[patched-fonts](https://github.com/cabrinha/nerd-fonts/tree/zed-fonts/patched-fonts)/[ZedMono](https://github.com/cabrinha/nerd-fonts/tree/zed-fonts/patched-fonts/ZedMono)/[Extended](https://github.com/cabrinha/nerd-fonts/tree/zed-fonts/patched-fonts/ZedMono/Extended) (I only download the Extended)
3. I copied the URL and went to a [third-party app to download the directory](https://download-directory.github.io/?url=https%3A%2F%2Fgithub.com%2Fcabrinha%2Fnerd-fonts%2Ftree%2Fzed-fonts%2Fpatched-fonts%2FZedMono%2FExtended).
4. Installed the fonts.
5. And just put them on VSCode and my Terminal.
6. Notice how on VSCode, I'm only using `"ZedMono Nerd Font"` and not `"ZedMono Nerd Font Extended"`. A little weird yeah but I guess it helps that I didn't really install the regular, thinner patched Zed Mono font? If I would have, it might have the same result as yours @arlyon.

**Comment by Finii** [comment]
> @Finii let me know if there are any changes I need to make here.

Thanks for the PR!

Well, what is special with this font, as we have some Iosevkas already?

image

arlyon commented:
> I downloaded this and don't see any difference between iosevka.

And
> So it works. Even works for the native MacOS Terminal (It has no fallback fonts unlike VSCode). Also got it to work on VSCode.

Better be! If the patch would not work that would be a bug in the Nerd Font patcher that needs fixing :->

This font seems to have some interest, but to consider it someone should point out the advantages/differences to a regular Iosevka.

Btw 'extended', iirc there is a feature request to include some Iosevka extended for the same reason as Blankeos'.

* #1475

**Comment by Blankeos** [comment]
@cabrinha I'm getting this issue with the font currently. Italic variant of `ZedMonoNF-Extended` doesn't seem to use `ZedMonoNF-ExtendedItalic`.

It seems to use `ZedMonoNF-Italic` instead.

### ⚠️ The issue:
image

### ✅ Ideally italics should look like this when using `ZedMonoNF-Extended`
image

**Comment by Finii** [comment]
Zed seems to be a bit more roundish, at least compared to our regular Iosevka:

image

**Comment by Finii** [comment]
Rebase on master, force push.

There are some errors in the font database file `fonts.json`, fixed en passant.

Are really all these versions needed?
* Maybe drop `Oblique`?
* Maybe drop some weights (see image below)
  * Thin (who can use that in a terminal?)
  * ExtraLight
  * SemiBold
  * Heavy

image

**Comment by Finii** [comment]
@Blankeos

> But actually it should be **Unique Name** that I'd use

Well, you need to specify the **Family Name** of the font, as it says in the option.

**Comment by Finii** [comment]
Upstream has not even the version right in the font file :sob:
This is supposed to be `1.2.0`?

One times, work with professionals...

image
image

File: bin/scripts/lib/fonts.json [hunk 1]
```diff
@@ -831,6 +831,18 @@
...
       "repoRelease": true,
       "description": "Clean, crisp and narrow, with a large x-height and clear punctuation"
+    },
+    {
+      "unpatchedName": "Zed Mono",
+      "RFN": false,
+      "patchedName": "ZedMono",
+      "folderName": "ZedMono",
+      "imagePreviewFont": "ZedMono Nerd Font",
+      "imagePreviewFontSource": "ZedMono/ZedMonoNerdFont-Regular.ttf",
```
Comment (Finii) [inline bin/scripts/lib/fonts.json:841]: Imagepreviewfont needs to point to the unpatched font.

**Comment by Finii** [comment]
Unfortunately we got a Cask clash...

* https://github.com/zed-industries/zed-fonts/issues/10#issuecomment-2028052925

This is when people are too impatient, sigh.

**Comment by Finii** [comment]
```
======== NotoSansMono_Condensed-ExtraBold.ttf ========
SFNT Fullname      ID 4     Noto Sans Mono Condensed ExtraBold
SFNT Family        ID 1     Noto Sans Mono Condensed ExtraBold
SFNT SubFamily     ID 2     Regular
SFNT Pref Family   ID 16    Noto Sans Mono Condensed
SFNT Pref Styles   ID 17    ExtraBold
SFNT PS Name       ID 6     NotoSansMonoCondensed-ExtraBold
```
```
======== zed-mono-extendedextrabolditalic.ttf ========
SFNT Fullname      ID 4     Zed Mono Extrabold Extended Italic
SFNT Family        ID 1     Zed Mono Extrabold Extended
SFNT SubFamily     ID 2     Italic
SFNT Pref Family   ID 16    Zed Mono
SFNT Pref Styles   ID 17    Extrabold Extended Italic
SFNT PS Name       ID 6     Zed-Mono-Extrabold-Extended-Italic
```

Hmm.

```python
    known_widths = { # can take modifiers
        'Compressed': ('Cm', 'Comp'),
        'Extended': ('Ex', 'Extd'),
        'Condensed': ('Cn', 'Cond'),
        'Narrow': ('Nr', 'Narrow'),
        'Compact': ('Ct', 'Compact'),
    }
```

All other fonts we patch that have a width specifier have that width first in their name.
Only Zed has not.
Adding the sorting changes the name for Zed and nor for any other. :+1: 

Examples
* `../src/unpatched-fonts/Noto/Serif/NotoSerif_ExtraCondensed-LightItalic.ttf`
  SFNT Fullname      ID 4     `Noto Serif ExtraCondensed Light Italic`
  SFNT PS Name       ID 6     `NotoSerifExtraCondensed-LightItalic`
* `../src/unpatched-fonts/Noto/Sans-Mono/NotoSansMono_SemiCondensed-ExtraLight.ttf`
  SFNT Fullname      ID 4     `Noto Sans Mono SemiCondensed ExtraLight`
  SFNT PS Name       ID 6     `NotoSansMonoSemiCondensed-ExtraLight`
* `../src/unpatched-fonts/3270/Condensed/3270Condensed-Regular.ttf`
  SFNT Fullname      ID 4     `IBM 3270 Condensed`
  SFNT PS Name       ID 6     `3270Condensed`
* `../src/unpatched-fonts/MartianMono/Cn/MartianMono-CondensedRegular.ttf`
  SFNT Fullname      ID 4     `Martian Mono Cn Rg`
  SFNT PS Name       ID 6     `MartianMono-CnRg`
* `../src/unpatched-fonts/Ubuntu/Condensed/Ubuntu-C_modified.ttf`
  SFNT Fullname      ID 4     `Ubuntu Condensed`
  SFNT PS Name       ID 6     `UbuntuCondensed-Regular`

Counterexample
* `../src/unpatched-fonts/ZedMono/Extended/zed-mono-extendedextraboldoblique.ttf`
  SFNT Fullname      ID 4     `Zed Mono Extrabold Extended Italic`
  SFNT PS Name       ID 6     `Zed-Mono-Extrabold-Extended-Italic`

**Comment by Finii** [comment]
image

:sob:

**Pull request merged**